### PR TITLE
Address admission cont default, QOS, and move to performance features.

### PR DESCRIPTION
**Pull request by stbof**
https://cockroachlabs.atlassian.net/browse/DOC-2726
https://cockroachlabs.atlassian.net/browse/DOC-3106
https://cockroachlabs.atlassian.net/browse/DOC-3105

### CODE REVIEW

**Review by sumeerbhola — CHANGES_REQUESTED** [review]
> Reviewed 6 of 9 files at r1, 2 of 4 files at r2, 1 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13632)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @stbof and @vy-ton)

___
*[v21.2/architecture/admission-control.md, line 11 at r3](https://reviewable.io/reviews/cockroachdb/docs/13632#-N00z9274V--yVuhvNzz:-N00z9274V--yVuhvO--:b-3tne9z) ([raw file](https://github.com/cockroachdb/docs/blob/90e7d3c61c9fcfd9daf5b5b5af19444aabfdc80f/v21.2/architecture/admission-control.md#L11)):*
> ```Markdown
> 
> {{site.data.alerts.callout_info}}
> Admission control is not available for {{ site.data.products.serverless }} clusters.
> ```

Do we need to document what we do on the multi-tenant storage servers (that run the KV and storage layer)? They are not under the control of the tenants, and it is likely that we will enable admission control on them in the very near future. I think what we do there ought to be of no concern of a tenant since this is "serverless". If we agree on that, this should be qualified to say that the per-tenant sql-only servers do not support admission control.
___
*[v22.1/admission-control.md, line 55 at r3](https://reviewable.io/reviews/cockroachdb/docs/13632#-N01-mQxC8919ha93Aq5:-N01-mQxC8919ha93Aq6:bxb0xob) ([raw file](https://github.com/cockroachdb/docs/blob/90e7d3c61c9fcfd9daf5b5b5af19444aabfdc80f/v22.1/admission-control.md#L55)):*
> ```Markdown
> The transaction start time is used within the priority queue and gives preference to operations with earlier transaction start times. For example, within the high priority queue operations with an earlier transaction start time are processed first.
> 
> ### Set operation priority for a session
> ```

Is this the only section that changed when moving this file from the old location?
Keeping it in the original location until the final PR merge would have shown the diff.

**Comment by stbof** [comment]
> > Reviewed 6 of 9 files at r1, 2 of 4 files at r2, 1 of 2 files at r3.
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13632)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @stbof and @vy-ton)
> 
> _[v21.2/architecture/admission-control.md, line 11 at r3](https://reviewable.io/reviews/cockroachdb/docs/13632#-N00z9274V--yVuhvNzz:-N00z9274V--yVuhvO--:b-3tne9z) ([raw file](https://github.com/cockroachdb/docs/blob/90e7d3c61c9fcfd9daf5b5b5af19444aabfdc80f/v21.2/architecture/admission-control.md#L11)):_
> 
> > ```gfm
> > {{site.data.alerts.callout_info}}
> > Admission control is not available for {{ site.data.products.serverless }} clusters.
> > ```
> 
> Do we need to document what we do on the multi-tenant storage servers (that run the KV and storage layer)? They are not under the control of the tenants, and it is likely that we will enable admission control on them in the very near future. I think what we do there ought to be of no concern of a tenant since this is "serverless". If we agree on that, this should be qualified to say that the per-tenant sql-only servers do not support admission control.

We don't discuss multi-tenant storage servers or per-tenant sql-only servers anywhere in the docs so I don't think it's appropriate to bring it up here. All end users know about is Dedicated and Serverless.

> 
> _[v22.1/admission-control.md, line 55 at r3](https://reviewable.io/reviews/cockroachdb/docs/13632#-N01-mQxC8919ha93Aq5:-N01-mQxC8919ha93Aq6:bxb0xob) ([raw file](https://github.com/cockroachdb/docs/blob/90e7d3c61c9fcfd9daf5b5b5af19444aabfdc80f/v22.1/admission-control.md#L55)):_
> 
> > ```gfm
> > The transaction start time is used within the priority queue and gives preference to operations with earlier transaction start times. For example, within the high priority queue operations with an earlier transaction start time are processed first.
> > 
> > ### Set operation priority for a session
> > ```
> 
> Is this the only section that changed when moving this file from the old location? Keeping it in the original location until the final PR merge would have shown the diff.

Noted. However, I wanted the PR to show what the docs looked like when the page is moved to the performance section.

**Review by vy-ton — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13632)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @stbof and @sumeerbhola)

___
*[v21.2/architecture/admission-control.md, line 11 at r3](https://reviewable.io/reviews/cockroachdb/docs/13632#-N00z9274V--yVuhvNzz:-N03ePsE5FqGakMOrJrD:b3t6qne) ([raw file](https://github.com/cockroachdb/docs/blob/90e7d3c61c9fcfd9daf5b5b5af19444aabfdc80f/v21.2/architecture/admission-control.md#L11)):*
I propose we remove this callout altogether. How we've built admission control for Serverless should not impact the user.

> it is likely that we will enable admission control on them in the very near future

When this happens, I think what we want to mention about Serverless is that tenants are protected from each other.
___
*[v22.1/admission-control.md, line 44 at r3](https://reviewable.io/reviews/cockroachdb/docs/13632#-N03gJNo21dKu4SmEfn5:-N03gJNo21dKu4SmEfn6:bn4u5af) ([raw file](https://github.com/cockroachdb/docs/blob/90e7d3c61c9fcfd9daf5b5b5af19444aabfdc80f/v22.1/admission-control.md#L44)):*
> ```Markdown
> 
> {{site.data.alerts.callout_info}}
> If you are upgrading to {{ page.version.version }}, first complete the upgrade with admission control disabled, then enable admission control after verifying the upgrade was successful.
> ```

This callout only applies to admission control in 21.2 not 22.1 since it is superseded by `Admission control is enabled by default for all layers.`
___
*[v22.1/admission-control.md, line 55 at r3](https://reviewable.io/reviews/cockroachdb/docs/13632#-N03gdC66oR4qaLw-8uE:-N03gdC66oR4qaLw-8uF:bfwy07) ([raw file](https://github.com/cockroachdb/docs/blob/90e7d3c61c9fcfd9daf5b5b5af19444aabfdc80f/v22.1/admission-control.md#L55)):*
> ```Markdown
> The transaction start time is used within the priority queue and gives preference to operations with earlier transaction start times. For example, within the high priority queue operations with an earlier transaction start time are processed first.
> 
> ### Set operation priority for a session
> ```

`Set quality of service level for a session` 

Can we replace `priority` with `service level` in this section? We're trying to not confuse users with transaction priorites

We should provide some background/context on why a user would need this feature. Something that conveys `In an overload scenario where CockroachDB cannot service all requests, users want control over what requests are prioritized, often referred to as quality of service. Admission control queues work throughout the system, and users can indicate more important work.`

**Comment by stbof** [comment]
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13632)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @stbof and @sumeerbhola)
> 
> _[v21.2/architecture/admission-control.md, line 11 at r3](https://reviewable.io/reviews/cockroachdb/docs/13632#-N00z9274V--yVuhvNzz:-N03ePsE5FqGakMOrJrD:b3t6qne) ([raw file](https://github.com/cockroachdb/docs/blob/90e7d3c61c9fcfd9daf5b5b5af19444aabfdc80f/v21.2/architecture/admission-control.md#L11)):_ I propose we remove this callout altogether. How we've built admission control for Serverless should not impact the user.
> 
> > it is likely that we will enable admission control on them in the very near future
> 
> When this happens, I think what we want to mention about Serverless is that tenants are protected from each other.
> 
> _[v22.1/admission-control.md, line 44 at r3](https://reviewable.io/reviews/cockroachdb/docs/13632#-N03gJNo21dKu4SmEfn5:-N03gJNo21dKu4SmEfn6:bn4u5af) ([raw file](https://github.com/cockroachdb/docs/blob/90e7d3c61c9fcfd9daf5b5b5af19444aabfdc80f/v22.1/admission-control.md#L44)):_
> 
> > ```gfm
> > {{site.data.alerts.callout_info}}
> > If you are upgrading to {{ page.version.version }}, first complete the upgrade with admission control disabled, then enable admission control after verifying the upgrade was successful.
> > ```
> 
> This callout only applies to admission control in 21.2 not 22.1 since it is superseded by `Admission control is enabled by default for all layers.`
> 
> _[v22.1/admission-control.md, line 55 at r3](https://reviewable.io/reviews/cockroachdb/docs/13632#-N03gdC66oR4qaLw-8uE:-N03gdC66oR4qaLw-8uF:bfwy07) ([raw file](https://github.com/cockroachdb/docs/blob/90e7d3c61c9fcfd9daf5b5b5af19444aabfdc80f/v22.1/admission-control.md#L55)):_
> 
> > ```gfm
> > The transaction start time is used within the priority queue and gives preference to operations with earlier transaction start times. For example, within the high priority queue operations with an earlier transaction start time are processed first.
> > 
> > ### Set operation priority for a session
> > ```
> 
> `Set quality of service level for a session`
> 
> Can we replace `priority` with `service level` in this section? We're trying to not confuse users with transaction priorites
> 
> We should provide some background/context on why a user would need this feature. Something that conveys `In an overload scenario where CockroachDB cannot service all requests, users want control over what requests are prioritized, often referred to as quality of service. Admission control queues work throughout the system, and users can indicate more important work.`

@vy-ton According to https://cockroachlabs.atlassian.net/browse/CRDB-7811, admission control is not yet supported for Serverless clusters. Are you suggesting that I still remove this:

{{site.data.alerts.callout_info}}
Admission control is not available for {{ site.data.products.serverless }} clusters.
{{site.data.alerts.end}}

**Review by vy-ton — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13632)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @stbof and @sumeerbhola)

___
*[`v22.1/admission-control.md` line 55 at r3](https://reviewable.io/reviews/cockroachdb/docs/13632#-N03gdC66oR4qaLw-8uE:-N0fciy6EiHAcg1gLVaD:b-mqhzcq) ([raw file](https://github.com/cockroachdb/docs/blob/90e7d3c61c9fcfd9daf5b5b5af19444aabfdc80f/v22.1/admission-control.md#L55)):*
<details><summary><i>Previously, stbof (Stephanie Bodoff) wrote…</i></summary><blockquote>

@vy-ton According to https://cockroachlabs.atlassian.net/browse/CRDB-7811, admission control is not yet supported for Serverless clusters. Are you suggesting that I still remove this:

{{site.data.alerts.callout_info}}
Admission control is not available for {{ site.data.products.serverless }} clusters.
{{site.data.alerts.end}}

</blockquote></details>

Ok, let's leave the call out for now. This should change with Serverless GA

**Review by vy-ton — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13632)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @stbof and @sumeerbhola)

___
*[`v22.1/admission-control.md` line 28 at r3](https://reviewable.io/reviews/cockroachdb/docs/13632#-N0fd_GQ2Kz8uXaaMgJL:-N0fd_GQ2Kz8uXaaMgJM:b-l2rv1z) ([raw file](https://github.com/cockroachdb/docs/blob/90e7d3c61c9fcfd9daf5b5b5af19444aabfdc80f/v22.1/admission-control.md#L28)):*
> ```Markdown
> 
> {{site.data.alerts.callout_info}}
> You should use admission control when overall cluster health is good but some nodes are experiencing overload. If you see these overload scenarios on many nodes in the cluster, that typically means the cluster needs more resources.
> ```

This callout no longer applies in 22.1 when admission control is on by default.

Maybe we can reword it to note `Admission control is beneficial when overall cluster health is good...`

**Review by vy-ton — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13632)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @stbof and @sumeerbhola)

___
*[`v22.1/admission-control.md` line 71 at r4](https://reviewable.io/reviews/cockroachdb/docs/13632#-N0kkp660mg6kuwW135B:-N0kkp660mg6kuwW135C:bvrd74j) ([raw file](https://github.com/cockroachdb/docs/blob/9a00f58fb92735828f64132e60f4be80e42cab67/v22.1/admission-control.md#L71)):*
> ```Markdown
> ~~~
> 
> To reset the priority to the default session setting (in between background and critical), run:
> ```

How are the settings on https://www.cockroachlabs.com/docs/dev/set-vars.html#supported-variables determined? We should mention `default_transaction_quality_of_service`

**Comment by stbof** [comment]
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13632)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @stbof and @sumeerbhola)
> 
> _[`v22.1/admission-control.md` line 71 at r4](https://reviewable.io/reviews/cockroachdb/docs/13632#-N0kkp660mg6kuwW135B:-N0kkp660mg6kuwW135C:bvrd74j) ([raw file](https://github.com/cockroachdb/docs/blob/9a00f58fb92735828f64132e60f4be80e42cab67/v22.1/admission-control.md#L71)):_
> 
> > ```gfm
> > ~~~
> > 
> > To reset the priority to the default session setting (in between background and critical), run:
> > ```
> 
> How are the settings on https://www.cockroachlabs.com/docs/dev/set-vars.html#supported-variables determined? We should mention `default_transaction_quality_of_service`

Thanks for that catch! I'll update that page.

**Review by rmloveland — APPROVED** [review]
:lgtm: 

> Reviewed 1 of 9 files at r1, 1 of 1 files at r4, 10 of 10 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13632#-:-N15FVUl-F9uWU6ahjoG:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @stbof, @sumeerbhola, and @vy-ton)

___
*[`_includes/v22.1/misc/session-vars.html` line 124 at r5](https://reviewable.io/reviews/cockroachdb/docs/13632#-N15EWlL475L9iahldpQ:-N15EWlL475L9iahldpR:b-anutjc) ([raw file](https://github.com/cockroachdb/docs/blob/72ab4d5e78f00082dd3ee7c8ff7f71baf7e6f97a/_includes/v22.1/misc/session-vars.html#L124)):*
> ```HTML
>     <code>default_transaction_quality_of_service</code>
>    </td>
>    <td><span class="version-tag">New in v22.1:</span> The default transaction quality of service for the current session.
> ```

It would be good to add a link from this description to the 'Set quality of service level' section on the admission control page so folks can read further about how to use this and get more context on it
